app/src/main/java/campus/tech/kakao/contacts/ContactsListActivity.kt

```diff
@@ -0,0 +1,136 @@
+            Contact(
+                "권성찬",
+                "010123123",
+                "kscksc@ksc.com",
+                AddContactActivity.GENDER_MALE,
+                LocalDate.now(),
+                "Hello!!"
+            )
+        )
+        addNewContact(Contact("장진욱", "4442358", null, null, null, null))
+    }
+
+    private fun initiateRecyclerView() {
+        recyclerView = findViewById(R.id.contact_list_recyclerView)
+        recyclerView.adapter =
+            ContactListAdapter(LayoutInflater.from(this), this, contactList) { _, index ->
+                clickContactItem(contactList[index])
+            }
+        recyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+    }
+
+    private fun initiateAddContactButton() {
+        val addContactButton = findViewById<Button>(R.id.button_add_contact)
+        addContactButton.setOnClickListener {
+            startAddContactActivity()
+        }
+    }
+
+    private fun addNewContact(contact: Contact) {
+        contactList.add(contact)
+        recyclerView.adapter?.notifyItemInserted(contactList.size)
+
+        setHelpMessageActive(false)
+    }
+
+    private fun removeContact(contact: Contact) {
+        contactList.remove(contact)
+
+        if (contactList.size == 0) {
+            setHelpMessageActive(true)
+        }
+    }
+
+    private fun startAddContactActivity() {
+        val intent = Intent()
+        val componentName = ComponentName(
+            this@ContactsListActivity,
+            AddContactActivity::class.java
+        )
+        intent.component = componentName
+        activityLauncher.launch(intent)
+    }
+
+    private fun startShowContactInfoActivity(contact: Contact) {
+        val intent = Intent()
+        val componentName = ComponentName(
+            this@ContactsListActivity,
+            ContactInfoActivity::class.java
+        )
+        intent.component = componentName
+        intent.putExtra("contact", contact)
+        startActivity(intent)
+    }
+
+    @Suppress("DEPRECATION")
+    private fun getContactFromIntent(intent: Intent?): Contact? {
+        if (intent == null)
+            return null
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            intent.getSerializableExtra(
+                AddContactActivity.KEY_CONTACT,
+                Contact::class.java
+            ) as Contact
+        } else {
+            intent.getSerializableExtra(AddContactActivity.KEY_CONTACT) as Contact?
+        }
+    }
+
+    private fun clickContactItem(contactItem: Contact) {
+        Log.d(
+            "KSC",
+            "Name: ${contactItem.name}, Phone: ${contactItem.phoneNumber}\nBirthday: ${contactItem.birthday ?: "NULL"},Email: ${contactItem.email ?: "NULL"}"
+        )
+        startShowContactInfoActivity(contactItem)
+    }
+
+    private fun setHelpMessageActive(active: Boolean) {
+        helpMessageText.visibility = if (active) View.VISIBLE else View.GONE
```

Comment:
안드로이드는 Visibility 관련 확장함수를 제공합니다.
```suggestion
        helpMessageText.isVisible = active
```
조금 더 직관적으로 표현할 수 있겠네요!

Reply:
수정하였습니다!